src/twisted/runner/procmon.py

```diff
@@ -241,7 +246,8 @@ def startProcess(self, name):
 
         args, uid, gid, env = self.processes[name]
 
-        proto = LoggingProtocol()
+        if proto is None:
+            proto = LoggingProtocol()
         proto.service = self
         proto.name = name
```

Comment:
Rather than requiring that attributes be mutated post-construction (this is *certainly* an antipattern!), a better pattern might be to allow passing a _callable_ that produces a `LoggingProtocol` instance?